app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionFragment.java

```diff
@@ -330,6 +359,63 @@ public void selected(ChannelInfoItem selectedItem) {
         importExportListHeader.setOnClickListener(v -> importExportOptions.switchState());
     }
 
+    private void showLongTapDialog(ChannelInfoItem selectedItem) {
+        final Context context = getContext();
+        final Activity activity = getActivity();
+        if (context == null || context.getResources() == null || getActivity() == null) return;
+
+        final String[] commands = new String[]{
+                context.getResources().getString(R.string.share),
+                context.getResources().getString(R.string.unsubscribe)
+        };
+
+        final DialogInterface.OnClickListener actions = (dialogInterface, i) -> {
+            switch (i) {
+                case 0:
+                    shareChannel(selectedItem);
+                    break;
+                case 1:
+                    deleteChannel(selectedItem);
+                    break;
+                default:
+                    break;
+            }
+        };
+
+        final View bannerView = View.inflate(activity, R.layout.dialog_title, null);
+        bannerView.setSelected(true);
+
+        TextView titleView = bannerView.findViewById(R.id.itemTitleView);
+        titleView.setText(selectedItem.getName());
+
+        TextView detailsView = bannerView.findViewById(R.id.itemAdditionalDetails);
+        detailsView.setVisibility(View.GONE);
+
+        new AlertDialog.Builder(activity)
+                .setCustomTitle(bannerView)
+                .setItems(commands, actions)
+                .create()
+                .show();
+
+    }
+
+    private void shareChannel (ChannelInfoItem selectedItem) {
+        shareUrl(selectedItem.getName(), selectedItem.getUrl());
+    }
+
+    @SuppressLint("CheckResult")
+    private void deleteChannel (ChannelInfoItem selectedItem) {
+        ExtractorHelper.getChannelInfo(selectedItem.getServiceId(), selectedItem.getUrl(), true)
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.newThread())
+                .subscribe((@NonNull ChannelInfo result) -> {
+                    List<SubscriptionEntity> toDelete = subscriptionService.subscriptionTable()
+                            .getSubscription(result.getServiceId(), result.getUrl())
+                            .blockingFirst();
```

Comment:
By convention, the code blocks in `subscribe` should run on `mainThread` since that is the only thread where UI changes can happen. I know that the UI changes for subscriptions occur on a different rx subscription, but this having `subscribe` run main thread is the convention and what devs usually expect it to be. 

What you want to do is chain a `onNext` after `subscribeOn` to run the deletion on io thread, and then use `observeOn` to switch back to main thread to subscribe, perhaps with a toast (which is UI thread only).